pkg/types/types.go

```diff
@@ -196,6 +202,11 @@ type E2EClusterProvider interface {
 	// using the cluster provider native way
 	GetKubeconfig() string
 
+	// GetLiveKubeconfig provides a way to extract the kubeconfig file associated with the cluster in question.
+	// Different from GetKubeconfig, this method is used to extract the kubeconfig live, where the GetKubeconfig is cached.
+	// It is useful in cases where the kubeconfig file is required to be extracted multiple times during the test run with different arguments.
+	GetLiveKubeconfig() (string, error)
```

Comment:
Not clear what the difference is. between `GetKubeConfig` and `GetLiveKubeConfig`  . It looks like in each implementation, it's just a call to `c.getKubeConfig`.

Reply:
I think I can remove this. I needed this at the start of the year (wrapper around same interface outside this code), but now use-case is slightly different and I will remove it.

Reply:
Ok, I see the problem now (still there)
```
func (c *Cluster) GetKubeconfig() string {
	return c.kubeConfigFile
}

func (c *Cluster) GetLiveKubeconfig() (string, error) {
	return c.getKubeConfig()
}
```
Basically `GetKubeconfig` always returns kubeconfig file, created during setup in tmp dir, where live will re-run get command and potentially re-apply args (might be changed with With helper)

So either we need to stop returning cached kubeconfig and always get fresh or still have this method :/

Reply:
`Latest` it is :) will update